pkg/operator/controller/route-metrics/controller.go

```diff
@@ -0,0 +1,180 @@
+package routemetrics
+
+import (
+	"context"
+	"fmt"
+
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+)
+
+const (
+	controllerName = "route_metrics_controller"
+)
+
+var (
+	log = logf.Logger.WithName(controllerName)
+)
+
+// New creates the route metrics controller. This is the controller
+// that handles all the logic for gathering and exporting
+// metrics related to route resources.
+func New(mgr manager.Manager, namespace string) (controller.Controller, error) {
+	// Create a copy of the cluster, but with a new cache to watch on Route objects from every namespace.
+	newCluster, err := cluster.New(mgr.GetConfig(), func(o *cluster.Options) {
```

Comment:
Can you accomplish a global namespace watch without adding a new controller-runtime cluster?  The cluster pattern is not used elsewhere in cluster-ingress-operator and I'm not certain it is needed for metrics collection.

Reply:
We would need a separate cache for watching all the namespaces. A new cluster by default creates a new cache as part of it. I'll check if only creating a new cache can work in this case.

Reply:
If you create a new cache instead of a cluster, you should be able to do this:

        mgr.Add(newCache)
	// ...
	if err := c.Watch(source.NewKindWithCache(&routev1.Route{}, newCache),

Reply:
I tried with a new cache and it worked fine. I'll push the changes.